### PR TITLE
Issue 664: Updating go version to 1.20

### LINKED ISSUE #664: Update go version in pravega-operator
### Description
Update the go version in pravega-operator
### Importance

should-have

### Location

Dockerfile

### Suggestions for an improvement

Update go version to 1.20

### PR DESCRIPTION
**Pull request by anishakj**
### Change log description

Updated go version to 1.20
### Purpose of the change

 Fixes #664

### What the code does

Changed the go version to 1.20

### How to verify it

E2E tests are passed

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -8,8 +8,8 @@
 #     http://www.apache.org/licenses/LICENSE-2.0
 #
 ARG DOCKER_REGISTRY
-ARG GO_VERSION=1.18
-ARG ALPINE_VERSION=3.16
+ARG GO_VERSION=1.20
```
Comment (jkhalack) [inline Dockerfile:11]: Could you please verify that 1.20 image has at least 1.20.3 golang version? There are quite a few CVE present in 1.20.0 that are fixed in 1.20.3.

Comment (anishakj) [reply Dockerfile:11]: It is having 1.20.5 
```
docker run -it golang:1.20-alpine3.18 /bin/sh
/go # go version
go version go1.20.5 linux/amd64
/go # 
```

**Review by a6dulaleem — APPROVED** [review]
LGTM

**Review by krishan-Rai — APPROVED** [review]
LGTM

**Pull request merged**